src-java/base-topology/base-messaging/src/test/java/org/openkilda/messaging/info/switches/v2/GroupsValidationEntryV2Test.java

```diff
@@ -0,0 +1,207 @@
+/* Copyright 2022 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.messaging.info.switches.v2;
+
+import static org.junit.Assert.assertEquals;
+
+import org.openkilda.messaging.info.switches.v2.GroupInfoEntryV2.BucketEntry;
+
+import com.google.common.collect.Lists;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GroupsValidationEntryV2Test {
+
+    @Test
+    public void splitAndUniteEmptyGroupEntryTest() {
+        GroupsValidationEntryV2 entry = GroupsValidationEntryV2.builder()
```

Comment:
Here and in test classes below:
I'd recommend to add inner class GroupsValidationEntryV2Factory, which would return test entries with meaningful names, e.g.
`GroupsValidationEntryV2Factory.getEmpty()` or `GroupsValidationEntryV2Factory.getHugeChunk()`
This will let you:
1) Increase test readability
2) Do not describe same test input twice.

Reply:
I'm not sure about readability. When you see name "empty" you do not know what does it mean (e.g. excess rules are null or empty list). What "huge" means? 10? 100? 1000? to check both names you will go to factory, at the end of the file, then you will go back to check chunk sizes and expected chunk count, then you will go to the end of file again to check if this huge object must be split on exactly X chunks or not.

At this moment test has 10 lines. You can see all of them in one screen. I think it has better readability.

Reply:
I'm sorry, but logic 'if it fits my screen size, breaking it would worsen readability' is wrong. Methods must be minimal logical units, not minimal screen size code units.

The readable naming rule for these particular test is quite simple, IMHO. What you have in test name is what your input name should be, e.g.:
```
@Test
void splitAndUniteEmptyGroupEntryTest() {
    expectedGroup = GroupsValidationEntryV2Factory.getEmpty()
...
@Test
void splitAndUniteGroupEntryWithMalformedObjectsTest() {
     expectedGroup = GroupsValidationEntryV2Factory.getGroupWithMalformedObjects()
... 
etc.
```
> What "huge" means? 10? 100? 1000? 
It doesn't matter, which amount of objects you defined as 'huge'. The goal is to understand what your test data tries to mimic (big array, corrupted archive, unchunkable amount of bytes, signal from Mars, whatever). Also it doesn't matter what you have in other fields of test data object, so you can hide these details under factory method name (if you don't want to describe a factory, make it Enum).